Enable deferred-decryption per default

th4s opened the pull request:
This PR enables deferred decryption per default. It exposes more fine-grained control over what size is decrypted locally after the TLS connection and what is decrypted online.

th4s commented on crates/prover/src/tls/config.rs:
```diff
@@ -100,11 +109,19 @@ impl ProverConfig {
 
     pub(crate) fn ot_sender_setup_count(&self) -> usize {
-        ot_send_estimate(Role::Prover, self.max_sent_data, self.max_recv_data)
+        ot_send_estimate(
```
yeah I think this would make sense, 2 questions:

1. Why is `EXTRA_OTS = 16384`? Is this the value for key exchange, prf and ghash?
2. Why is `ot_send_estimate` for `Prover` only `EXTRA_OTS`? Same for `ot_recv_estimate` and `Verifier`?

heeckhau commented on crates/prover/src/tls/config.rs:
```diff
@@ -22,11 +22,21 @@ pub struct ProverConfig {
-    #[builder(default = "0")]
+    #[builder(default = "DEFAULT_MAX_RECV_LIMIT")]
     max_deferred_size: usize,
+    /// Defers the decryption from the start of the connection.
```
```suggestion
    /// Defers decryption until after the connection is closed.
```

heeckhau commented on crates/prover/src/tls/config.rs:
```diff
@@ -22,11 +22,21 @@ pub struct ProverConfig {
+    /// This allows to decrypt responses locally without MPC, so this option saves bandwidth
+    /// and performance.
+    #[builder(default = "true")]
+    defer_decryption_from_start: bool,
```
I feel it would be better to refer to when it is deferred to (after close), instead of saying it is deferred at the start

th4s replied:
Hmm I explicitly added `from_start` because we can also enable it midflight, so that people do not wonder what this is for when we also have `defer_decryption()`.

th4s commented:
Currently experiencing #559 locally...

th4s commented:
> Currently experiencing #559 locally when testing this branch.

This did not have anything to do with #559 and was a bug I introduced, but is now fixed. Rdy for review.

th4s commented:
The rebase is turning into a rewrite so it is easier to open a new PR and redo this.
